source/Playnite/GamesEditor.cs

```diff
@@ -467,6 +467,206 @@ public void OpenGameLocation(Game game)
             }
         }
 
+        public void CalculateGameSize(Game game, bool onlyIfDataMissing, bool updateGame = true, bool checkLastScanDate = false)
+        {
+            if (!game.IsInstalled)
+            {
+                return;
+            }
+
+            if (game.InstallSize != null && onlyIfDataMissing)
+            {
+                return;
+            }
+
+            try
+            {
+                long size = 0;
+                if (game.Roms.HasItems())
+                {
+                    var expandedGame = GetExpandedGameForRomsSizeScan(game);
+                    if (checkLastScanDate)
+                    {
+                        var addedAfterLastCheck = expandedGame.LastSizeScanDate == null || expandedGame.Added != null && expandedGame.Added > expandedGame.LastSizeScanDate;
+                        if (!addedAfterLastCheck && !expandedGame.Roms.Any(x => !x.Path.IsNullOrWhiteSpace()
+                            && FileSystem.FileExists(x.Path) && expandedGame.LastSizeScanDate != null
+                            && FileSystem.FileGetLastWriteTime(x.Path) > expandedGame.LastSizeScanDate))
+                        {
+                            return;
+                        }
+                    }
+
+                    foreach (var rom in expandedGame.Roms)
+                    {
+                        if (rom.Path.IsNullOrWhiteSpace() || !FileSystem.FileExists(rom.Path))
+                        {
+                            continue;
+                        }
+
+                        if (rom.Path.EndsWith(".cue", StringComparison.OrdinalIgnoreCase))

```

Comment:
This m3u and cue handling is completely duplicated code. There should be only one implementation that just accepts list of files you get from playlist file parser.

Or ideally just move them to some temporary ROM list and treat them in the end as normally assigned rom files, which means that you will need just that last `else` branch for the size calculation.

Reply:
Done in https://github.com/JosefNemec/Playnite/pull/3017/commits/046a08dff6bffdf451441cf191364da80cbd526f